src/communication/network_interface_sr.c

```diff
@@ -10698,3 +10698,231 @@ scdc_end_session (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int
 
   return;
 }
+
+static int
+flashback_verify_time (THREAD_ENTRY * thread_p, time_t start_time, time_t end_time, LOG_LSA * start_lsa,
+		       LOG_LSA * end_lsa)
+{
+  int error_code = NO_ERROR;
+  time_t ret_time = 0;
+
+
+  /* 1. Check start_time */
+  if (start_time < log_Gl.hdr.db_creation)
+    {
+      return ER_FLASHBACK_INVALID_TIME;
+    }
+  else
+    {
+      ret_time = start_time;
+
+      error_code = cdc_find_lsa (thread_p, &ret_time, start_lsa);
+      if (error_code == NO_ERROR || error_code == ER_CDC_ADJUSTED_LSA)
+	{
+	  /* find log record at the time
+	   * ret_time : time of the commit record or dummy record that is found to be greater than or equal to the start_time at first */
+
+	  if (ret_time > end_time)
+	    {
+	      /* out of range : start_time (ret_time) can not be greater than end_time */
+	      return ER_FLASHBACK_INVALID_TIME;
+	    }
+	}
+      else
+	{
+	  /* failed to find a log at the time due to failure while reading log page or volume (ER_FAILED, ER_LOG_READ) */
+	  return ER_FAILED;
```

Comment:
What if it can't find the log record because the time is too large or small? It isn't `ER_FLASHBACK_INVALID_TIME`?

Reply:
validate if the time is too small with comparing `db_creation` .
and compare `end_time` to see if time is too large. 
end_time should be less than current time.

Reply:
OK!